rust/x509/src/x509_example_rust/tests/tests.rs

```diff
@@ -0,0 +1,271 @@
+                    .verify(
+                        &root_certificate
+                            .public_key()
+                            .expect("failed to get public key")
+                    )
+                    .expect("errors occurred while verifying child certificate"),
+                "failed to verify child certificate"
+            );
+        }
+    }
+}
+
+fn pic_and_canister_id(ca_key_information: CaKeyInformation) -> (PocketIc, Principal) {
+    let pic = PocketIcBuilder::new()
+        .with_application_subnet()
+        .with_ii_subnet()
+        .with_fiduciary_subnet()
+        .build();
+
+    let canister_id = pic.create_canister();
+    pic.add_cycles(canister_id, 2_000_000_000_000);
+
+    let wasm = std::fs::read(WASM_PATH).expect("Wasm file not found, run 'dfx build'.");
+
+    pic.install_canister(
+        canister_id,
+        wasm,
+        encode_one(ca_key_information).unwrap(),
+        None,
+    );
+
+    (pic, canister_id)
+}
+
+fn fetch_root_ca_certificate(pic: &PocketIc, canister_id: Principal) -> X509 {
+    let root_certificate_result: Result<X509CertificateString, String> = match pic
+        .update_call(
+            canister_id,
+            Principal::anonymous(),
+            "root_ca_certificate",
+            encode_one(()).unwrap(),
+        )
+        .expect("Failed to call counter canister")
+    {
+        WasmResult::Reply(r) => decode_one(&r).expect("failed to decode reply"),
+        WasmResult::Reject(r) => panic!("Call failed: {:?}", r),
+    };
+
+    let root_certificate_pem = root_certificate_result.expect("failed to compute root certificate");
+
+    X509::from_pem(root_certificate_pem.x509_certificate_string.as_bytes())
+        .expect("failed to decode root certificate")
+}
+
+fn generate_child_certificate(pem_req: Vec<u8>, pic: &PocketIc, canister_id: Principal) -> X509 {
+    let pem_certificate_request = String::from_utf8(pem_req).expect("invalid request encoding");
+    let child_certificate_result: Result<X509CertificateString, String> = match pic
+        .update_call(
+            canister_id,
+            Principal::anonymous(),
+            "child_certificate",
+            encode_one(PemCertificateRequest {
+                pem_certificate_request,
+            })
+            .unwrap(),
+        )
+        .expect("Failed to call counter canister")
+    {
+        WasmResult::Reply(r) => decode_one(&r).expect("failed to decode reply"),
+        WasmResult::Reject(r) => panic!("Call failed: {:?}", r),
+    };
+
+    let child_certificate_text =
+        child_certificate_result.expect("failed to compute child certificate");
+
+    X509::from_pem(child_certificate_text.x509_certificate_string.as_bytes())
+        .expect("failed to decode child certificate")
+}
+
+fn generate_child_certificate_request(
+    key: &PKeyRef<Private>,
+    digest_type: MessageDigest,
+) -> X509Req {
+    let mut builder = X509Name::builder().expect("failed to create X509NameBuilder");
```

Comment:
Just for my understanding: could we also have used [x509_cert](https://docs.rs/x509-cert/latest/x509_cert/index.html)::[builder](https://docs.rs/x509-cert/latest/x509_cert/builder/index.html)::[RequestBuilder](https://docs.rs/x509-cert/latest/x509_cert/builder/struct.RequestBuilder.html#) here instead of OpenSSL?